### ISSUE TITLE
Add wasm tests

### ISSUE THREAD

**Issue by max-sixty** [PRQL/prql#1983]
I only recently realized that we're not running any tests on wasm. The normal tests (with `#[test]`) are not run in the wasm target. We can see that in the test output at https://github.com/PRQL/prql/actions/runs/4288400460/jobs/7470357470

Ideally we'd do at least some basic tests. I don't think we have file access in wasm, but if we included some of the examples with `include_str!` and checked they compiled (even without checking the result), that would be good progress. My guess is that we're going to get failures in executing functions — for example our recent issues with stack space with Chumsky — rather than a subtlety different output.

**Comment by aljazerzen** [comment]
We do have tests for prql-js that run on node, so we do test the wasm module.

What exactly do you mean with "wasm tests"?

**Comment by max-sixty** [comment]
> What exactly do you mean with "wasm tests"?

I had meant the tests that run on:

```
cargo test --target=wasm32-unknown-unknown
```

> We do have tests for prql-js that run on node, so we do test the wasm module.

Yes, 100%, I should have included that in the message above. They're brief, but they do cover most likely failures.

Still — adding some basic tests that run from cargo could just be a couple of test functions which iterate over our examples, and cover a much wider range of failures. 

They'd also run in our rust tests, which run on every PR, so would gate any rust or rust dependency changes.

Does that make sense to you?

**Comment by aljazerzen** [comment]
We could that that with wasm-bindgen: https://rustwasm.github.io/docs/book/game-of-life/testing.html

**Comment by max-sixty** [comment]
> We could that that with wasm-bindgen: https://rustwasm.github.io/docs/book/game-of-life/testing.html

Yes we already have that setup it seems, we just don't run any tests...

https://github.com/PRQL/prql/blob/b05c626f9b6fb00aa18e7864e6b9b44e1a06a664/.cargo/config.toml#L1-L2
